ansible/VMSS/vmss_create.yml

```diff
@@ -0,0 +1,98 @@
+# Description
+# ===========
+# This playbook creates a Virtual Machine Scaleset.
+# It also requires following env variables to be defined:
+# - VMSS_ADMIN_USERNAME
+# - VMSS_ADMIN_PASSWORD
+# This sample requires Ansible 2.6 
+
+---
+- hosts: localhost
+  vars_files:
+  - vmss_vars.yml
+
+  tasks:
+    - name: Create a resource group
+      azure_rm_resourcegroup:
+        name: "{{ resource_group }}"
+        location: "{{ location }}"
+    - name: Create virtual network
+      azure_rm_virtualnetwork:
+        resource_group: "{{ resource_group }}"
+        name: "{{ vmss_name }}"
+        address_prefixes: "10.0.0.0/16"
+    - name: Add subnet
+      azure_rm_subnet:
+        resource_group: "{{ resource_group }}"
+        name: "{{ vmss_name }}"
+        address_prefix: "10.0.1.0/24"
+        virtual_network: "{{ vmss_name }}"
+    - name: Create public IP address
+      azure_rm_publicipaddress:
+        resource_group: "{{ resource_group }}"
+        allocation_method: Static
+        name: "{{ vmss_name }}"
+    - name: Create Network Security Group that allows SSH
+      azure_rm_securitygroup:
+        resource_group: "{{ resource_group }}"
+        name: "{{ vmss_name }}"
+        rules:
+          - name: SSH
+            protocol: Tcp
+            destination_port_range: 22
```

Comment:
why need open 22

Reply:
After the VMSS provisioning, vmss_deploy need to ssh to those VMs to do the deployment which requires port 22 to be opened.